tools/inference_engine/src/impl/model.h

```diff
@@ -85,18 +83,36 @@ inline INode* to_node(inference_engine_node_t n)
 
 struct Port : public INode
 {
-    Port(const inference_engine_port_desc_t& desc)
-        : INode(ModelNodeType::ePort, {})
-    {
-    }
+	Port(const inference_engine_port_desc_t& desc)
+		: INode(ModelNodeType::ePort, {})
+    , tensor_(desc.tensor)
+	{
+		type_ = ModelNodeType::ePort;
+		outputs_.push_back(this);
+	}
+
+	const Tensor& tensor() const
+	{
+		return tensor_;
+	}
+
+private:
+	Tensor tensor_;
 };
 
 struct MatMul : public INode
 {
-    MatMul(const inference_engine_matmul_desc_t& desc) 
-        : INode(ModelNodeType::eMatmul, { to_node(desc.tensor_a), to_node(desc.tensor_b) })
-    {
-    }
+	MatMul(const inference_engine_matmul_desc_t& desc);
+
+	const Tensor& tensor_a() const
+	{
+		return reinterpret_cast<Port*>(inputs_[0])->tensor();
```

Comment:
Casting to `Port*`, but a) what guarantees there are that inputs are ports? b) have we to expose this 2 methods (tensor_a(), tensor_b()) as public ones, at current stage of codebase?